### PR TITLE
Pro 5355 localization translation

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-5355](https://linear.app/apostrophecms/issue/PRO-5355/as-a-user-i-want-to-automatically-translate-content-when-localizing-it)

## Summary

* Create the new chip for translation
* Adds UI in the localization manager (with and without related docs)
* Handle different errors scenarios (mocked since no provider available)

## What are the specific steps to test this change?

Try to localize a document or page without related docs, just before to localize you should see this new section:
image

If you check the translate checkbox, you should see an error:
image

Errors for unavailable targets are already working but not functional since no provider right now.

You should be able to do the same with piece with related docs:
image
 

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/i18n/i18n/en.json [hunk 1]
```diff
@@ -45,6 +45,11 @@
   "assetWebpackCacheCleared": "Build cache cleared.",
   "assetWebpackConfigWarning": "⚠️ In the module {{ module }}, your webpack config is incorrect. It must be an object and should contain only the properties: {{ properties }}.",
   "at": "at",
+  "automaticTranslationCheckbox": "Translate text content",
+  "automaticTranslationSettings": "Automatic translation settings",
+  "automaticTranslationSourceErrMsg": "The current locale <strong>{{ source }}</strong> is not suitable for translation. Translation cannot happen.",
```
Comment (myovchev) [inline modules/@apostrophecms/i18n/i18n/en.json:50]: "Translation will be skipped" probably? Not sure though, native English speakers have to advise.

File: modules/@apostrophecms/i18n/i18n/en.json [hunk 2]
```diff
@@ -45,6 +45,11 @@
...
+  "automaticTranslationTargetErrMsg": "Could not find a suitable language for the locale <strong>{{ targets }}</strong>. Translation will be skipped when localizing.",
+  "automaticTranslationTargetErrMsg_plural": "Could not find a suitable language for the locales <strong>{{ targets }}</strong>. Translation will be skipped when localizing.",
```
Comment (myovchev) [inline modules/@apostrophecms/i18n/i18n/en.json:52]: What bothers me with those messages (coming from the prototype) is that it's really not clear that the translation will be skipped for the given locales only. At  least this is our plan. Again I think this should be synced with Stu and Tom probably.

File: modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue [hunk 3]
```diff
@@ -370,7 +395,9 @@ export default {
           value: 'relatedDocsOnly',
           label: 'apostrophe:relatedDocsOnly'
         }
-      ]
+      ],
+      translationEnabled: apos.modules['@apostrophecms/translation']?.enabled,
```
Comment (myovchev) [inline modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue:399]: I think optional chaining not needed anymore.

**Review by myovchev — CHANGES_REQUESTED** [review]
Tiny stuff.

File: modules/@apostrophecms/i18n/i18n/en.json (hunk 2 above)
Comment (ValJed) [reply modules/@apostrophecms/i18n/i18n/en.json]: Updated based on Tom and Stu feedback.

File: modules/@apostrophecms/i18n/i18n/en.json (hunk 1 above)
Comment (ValJed) [reply modules/@apostrophecms/i18n/i18n/en.json]: Right, I updated that.

File: modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue (hunk 3 above)
Comment (ValJed) [reply modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue]: Removed

**Review by myovchev — APPROVED** [review]

**Pull request merged**